File: 01/main.js
var fs = require('fs');

if (process.argv.length < 6) {
  return;
}

var algo = process.argv[2];
var op = process.argv[3];
var entrada = process.argv[4];
var chave = process.argv[5];

function gravaSaida(filename, codigo) {
  fs.open(filename, "w+", function(err, fd) {
    if (err) {
      return;
    }

    fs.write(fd, codigo, 0, codigo.length, function(err, written, buffer) {
      if (err) {
        return;
      }
    });
  });
}

fs.open(entrada, "r", function(err, f) {
  if (err) {
    return;
  }

  fs.readFile(entrada, function(err, data) {
    if (err) {
      return;
    }

    var algoritmo = require("./" + algo);
    algoritmo.chave(chave, function(a) {
      var codigo = algoritmo[op](data.toString(), a);

      var saida = algo + "_" + op + ".txt";
      gravaSaida(saida, codigo);
    });
  });
});

File: 01/main.py
from sys import argv, path
path.append("../cifras")
from cesar import Cesar
from transp import Transposicao
from vigenere import Vigenere
from substituicao import Substituicao

script, algo,op,entrada,chave = argv

def gravaSaida(filename, codigo):
  with open(filename, "w+v") as f:
    f.write(codigo)

with open(entrada, "r+b") as f:
  texto = ""

  while 1:
    c = f.read(1)

    if not c:
      break

    texto += c

  if algo == "cesar":
    obj = Cesar()
  elif algo == "transposicao":
    obj = Transposicao()
  elif algo == "vigenere":
    obj = Vigenere()
  elif algo == "substituicao":
    obj = Substituicao()

  chave = obj.chave(chave)

  if op == "cifra":
    codigo = obj.cifra(texto, chave)
  else:
    codigo = obj.decifra(texto, chave)

  gravaSaida(algo + "_" + op + ".txt", codigo)
